src/cswinrt/strings/ComInteropHelpers.cs

```diff
@@ -53,22 +53,35 @@
 
 namespace WinRT.Interop
 {
-    [ComImport]
-    [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
-    [Guid("EECDBF0E-BAE9-4CB6-A68E-9598E1CB57BB")]
-    internal interface IWindowNative
+ 
+#if EMBED
+    internal 
+#else
+    public
+#endif
+    static class IWindowNativeMethods
     {
-        IntPtr WindowHandle { get; }
+        public static unsafe global::System.IntPtr get_WindowHandle(object _obj)
+        {
+            var asObjRef = global::WinRT.MarshalInspectable<object>.CreateMarshaler2(_obj, System.Guid.Parse("EECDBF0E-BAE9-4CB6-A68E-9598E1CB57BB"));
+
+            var ThisPtr = asObjRef.GetAbi();
+
+            global::System.IntPtr __retval = default;
+            global::WinRT.ExceptionHelpers.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, out global::System.IntPtr, int>**)ThisPtr)[3](ThisPtr, out __retval));
+            return __retval;
+        }
     }
 
-#if EMBED 
-    internal 
-#else 
-    public 
+
+#if EMBED
+    internal
+#else
+    public
 #endif
     static class WindowNative
     {
-        public static IntPtr GetWindowHandle(object target) => target.As<IWindowNative>().WindowHandle;
+        public static IntPtr GetWindowHandle(object target) => IWindowNativeMethods.get_WindowHandle(target);
     }
 
     [ComImport]

```

Comment:
You will also need to address this interface as it is a ComImport interface.

Reply:
addressing separately